File: euler92.py
l89 = [0]*(7*81+1) 
l1 = [0]*(7*81+1)

def convergeto(n):
    global l1
    global l89
    b = (sum([int(k)**2 for k in str(n)]))
    while True:
        if b==1 or l1[b-1]:
            return 1
        elif b==89 or l89[b-1]:
            return 89
        b = sum([int(k)**2 for k in str(b)])


for x in range(1, 7*81+1):
    print x, convergeto(x)
    if convergeto(x)==1:
        l1[x-1]=1
    else:
        l89[x-1]=1

cnt = 0
for x in range(1, 10**7):
    s = sum([int(k)**2 for k in str(x)])
    if l89[s-1]:
        cnt+=1
print cnt

File: euler92.cpp
#include <iostream>
using namespace std;

int sqsum(long long n)
{
	int sum=0;
	int r, a;
	while(n!=0) {
		r = n%10;
		sum+=r*r;
		n/=10;
	}
	return sum;
}

int main()
{
	int cnt=0, temp;
	for(int i = 1; i < 10000000; i++)
	{
		temp=i;
		while(1){
			if (temp==1)break;
				if(temp==89) {
					cnt+=1;
					break;
				}
				temp=sqsum(temp);
		}
	}
	cout<< cnt;
}
